pkgs/stdenv/generic/setup.sh

```diff
@@ -1558,6 +1547,7 @@ genericBuild() {
             configurePhase ${preBuildPhases[*]:-} buildPhase checkPhase \
             ${preInstallPhases[*]:-} installPhase ${preFixupPhases[*]:-} fixupPhase installCheckPhase \
             ${preDistPhases[*]:-} distPhase ${postPhases[*]:-}";
+        echo "phases: $phases" | tr -s ' '
```

Comment:
This also shows the phases which have been disabled with `dontX` because the `dontX` logic is done in the `for curPhase` loop

Reply:
@imincik this doesn't seem resolved?

Reply:
Sorry, thanks for re-opening.

Reply:
@Artturin , what about [this change](https://github.com/NixOS/nixpkgs/pull/257229/commits/288756e6017c9e036e2f11863cff6a8c5d03680f).

* loop over the phases to get the list of those which are not disabled - `activePhases`
* print list of `activePhases`
* loop over the `activePhases` in order to execute them ?

Reply:
@Artturin , would you please review my changes ? Thank you very much.